### PR TITLE
BUG : Adding *kwargs for name retention in Index args

### LINKED ISSUE #21697: BUG: to_datetime with Index arg does not retain name in all cases
```
#Example

In [17]: idx = pd.Index([15e9], name='name')

In [18]: pd.to_datetime(idx, errors='ignore', box=True, unit='s').name

In [19]: pd.__version__
Out[19]: '0.24.0.dev0+193.gdc45fba'
```
There are certain cases where the name attribute of the passed `Index` are not retained after `to_datetime`.

The `name` kwarg looks to be missing from these locations:

https://github.com/pandas-dev/pandas/blob/dc45fbafef172e357cb5decdeab22de67160f5b7/pandas/core/tools/datetimes.py#L102

https://github.com/pandas-dev/pandas/blob/dc45fbafef172e357cb5decdeab22de67160f5b7/pandas/core/tools/datetimes.py#L139

https://github.com/pandas-dev/pandas/blob/dc45fbafef172e357cb5decdeab22de67160f5b7/pandas/core/tools/datetimes.py#L344

### PR DESCRIPTION
**Pull request by uds5501**
- [x] closes #21697
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by mroeschke — CHANGES_REQUESTED** [review]
Needs tests and whatsnew note (0.24.0.txt). Could you also modify this PR title to be a little more descriptive of the issue that this solves?

**Comment by uds5501** [comment]
@mroeschke  Sure sir, would do

**Comment by uds5501** [comment]
@mroeschke I had a quick question. I made a commit regarding the whatsnew note in another path. If I want to add the commit in this PR itself.. how do I do that?

**Comment by WillAyd** [comment]
@uds5501 I am not sure if I completely follow but you could potentially merge the other branch into this branch locally or just re-do the whatsnew entry. The latter is probably easiest since we are just talking about a one-liner

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -526,6 +526,119 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     pandas.DataFrame.astype : Cast argument to a specified dtype.
     pandas.to_timedelta : Convert argument to timedelta.
     """
+    from pandas.core.indexes.datetimes import DatetimeIndex
```
Comment (mroeschke) [inline pandas/core/tools/datetimes.py]: This block below already exists in `_convert_listlike_datetimes`

**Comment by jreback** [comment]
closing as dupe of #21848 which is farther along

**Pull request closed without merging**